arch/rv32i/src/epmp.rs

```diff
@@ -74,39 +74,70 @@ impl<const MAX_AVAILABLE_REGIONS_OVER_TWO: usize> PMP<MAX_AVAILABLE_REGIONS_OVER
         let mut num_regions = 0;
         let mut locked_region_mask = 0;
 
-        for i in 0..(MAX_AVAILABLE_REGIONS_OVER_TWO * 2) {
-            // Read the current value
-            let pmpcfg_og = csr::CSR.pmpconfig_get(i / 4);
-
-            // Flip R, W, X bits
-            let pmpcfg_new = pmpcfg_og ^ (3 << ((i % 4) * 8));
-            csr::CSR.pmpconfig_set(i / 4, pmpcfg_new);
-
-            // Check if the bits are set
-            let pmpcfg_check = csr::CSR.pmpconfig_get(i / 4);
-
-            // Check if the changes stuck
-            if pmpcfg_check == pmpcfg_og {
-                // If we get here then our changes didn't stick, let's figure
-                // out why
-
-                // Check if the locked bit is set
-                if pmpcfg_og & ((1 << 7) << ((i % 4) * 8)) > 0 {
-                    // The bit is locked. Mark this regions as not usable
-                    locked_region_mask |= 1 << i;
-                } else {
-                    // The locked bit isn't set
-                    // This region must not be connected, which means we have run out
-                    // of usable regions, break the loop
-                    break;
-                }
+        if csr::CSR.mseccfg.is_set(csr::mseccfg::mseccfg::mmwp) {
```

Comment:
nit: This function (`new`) should be renamed to something like `auto_detect`.

In the future, some implementations are going to want to pass in a per-board (or per-chip) static PMP configuration that represents the state of the PMP at init time and will instruct this implementation about pre-configured regions and free regions available for the OS and apps.